tests/test_2qbits_gates.py

```diff
@@ -0,0 +1,162 @@
+# MIT License
+#
+# Copyright (c) 2022 Quandela
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# As a special exception, the copyright holders of exqalibur library give you
+# permission to combine exqalibur with code included in the standard release of
+# Perceval under the MIT license (or modified versions of such code). You may
+# copy and distribute such a combined system following the terms of the MIT
+# license for both exqalibur and Perceval. This exception for the usage of
+# exqalibur is limited to the python bindings used by Perceval.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import pytest
+import numpy as np
+
+import perceval as pcvl
+from perceval import BasicState, generate_all_logical_states, SimulatorFactory, Processor
+from perceval.components import catalog, BS, get_basic_state_from_ports
+from perceval.algorithm import Analyzer
+
+
+def test_fidelity_and_performance_cnot():
+    # Tests the performance and the fidelity of different CNOT in perceval
+    ports = [pcvl.Port(pcvl.Encoding.DUAL_RAIL, "")] * 2
+    state_dict = {pcvl.components.get_basic_state_from_ports(ports, state): str(
+        state) for state in pcvl.utils.generate_all_logical_states(2)}
+    expected = {"00": "00", "01": "01", "10": "11", "11": "10"}
+
+    # KLM CNOT
+    analyzer_klm_cnot = Analyzer(catalog["klm cnot"].build_processor(), state_dict)
+    analyzer_klm_cnot.compute(expected=expected)
+
+    assert pytest.approx(analyzer_klm_cnot.fidelity, 1E-4) == 1
+
+    # Postprocessed CNOT
+    analyzer_postprocessed_cnot = Analyzer(catalog["postprocessed cnot"].build_processor(), state_dict)
+    analyzer_postprocessed_cnot.compute(expected=expected)
+
+    assert analyzer_postprocessed_cnot.fidelity == 1
```

Comment:
Don't you miss a `pytest.approx` here?

Reply:
No because our postprocessed_cnot is exacly one, we don't need the approx

Reply:
I just noticed that we are using the same method for all our different CNOT gates, maybe we can simplify by using `pytest.parametrize`?

Reply:
is the fidelity of the heralded_cnot also exactly = 1 ? or maybe we need pytest.approx here?